tools/create_operators_pull_request.py

```diff
@@ -0,0 +1,245 @@
+        default=False,
+        help="Show debug output from all operations performed",
+    )
+
+    args = parser.parse_args()
+    operator_repository = args.operator_repository
+    operator_name = args.operator_name
+    operator_git_tag = args.operator_git_tag
+    github_token = args.github_token
+    git_commit_message = args.git_commit_message or (
+        f"Release {operator_name} {operator_git_tag} (automated commit)."
+    )
+    operators_base_branch = args.operators_base_branch
+    git_user = args.git_user
+    debug = args.debug
+
+    base_directory = tempfile.mkdtemp("_kudo_operator_tools")
+    operators_repository = KUDOBUILDER_OPERATORS_REPOSITORY
+    operators_repository_url = (
+        f"{git_user}@github.com:{operators_repository}.git"
+    )
+    operators_branch = (
+        f"{operator_name}_{operator_git_tag}_{random_short_string()}"
+    )
+
+    rc, operators_directory, error_message = clone_repository(
+        operators_repository_url, operators_base_branch, base_directory, debug
+    )
+    if rc != 0:
+        log.error(error_message)
+        return rc
+
+    operator_repository_url = (
+        f"https://{git_user}:{github_token}@github.com"
+        + f"/{operator_repository}.git"
+    )
+
+    rc, operator_directory, error_message = clone_repository(
+        operator_repository_url, operator_git_tag, base_directory, debug
+    )
+    if rc != 0:
+        log.error(error_message)
+        return rc
+
+    rc, stdout, stderr = run(
+        f"cd {operators_directory} && git checkout -b {operators_branch}",
+        debug=debug,
+    )
+    if rc != 0:
+        return (
+            rc,
+            f"Failed to create kudobuilder/operators branch: "
+            + f"{operators_branch}\nstdout:\n{stdout}\nstderr:\n{stderr}",
+        )
+
+    rc, stdout, stderr = run(
+        f"cd {operator_directory} && git show -s --format='%an' {operator_git_tag}",
+        debug=debug,
+    )
+    if rc != 0:
+        return (
+            rc,
+            f"Failed to get git user name from "
+            + f"{operator_repository}@{operator_git_tag}"
+            + f"\nstdout:\n{stdout}\nstderr:\n{stderr}",
+        )
+    git_user_name = stdout.strip()
+
+    rc, stdout, stderr = run(
+        f"cd {operator_directory} && git show -s --format='%ae' {operator_git_tag}",
+        debug=debug,
+    )
+    if rc != 0:
+        return (
+            rc,
+            f"Failed to get git user email from "
+            + f"{operator_repository}@{operator_git_tag}"
+            + f"\nstdout:\n{stdout}\nstderr:\n{stderr}",
+        )
+    git_user_email = stdout.strip()
+
+    rc, error_message = configure_git_user(
+        operators_directory, git_user_name, git_user_email, debug
+    )
+    if rc != 0:
+        log.error(error_message)
+        return rc
+
+    versioned_operator_directory = (
+        f"{operators_directory}/repository/{operator_name}/{operator_git_tag}"
+    )
+
+    command = " && ".join(
```

Comment:
Can you use native Python filesystem operations rather than shelling out to `mkdir` and `cp`?

Reply:
Unless you feel strongly about it, I'll use [this argument](https://github.com/mesosphere/kudo-cassandra-operator/pull/26#discussion_r351832646) to justify not spending too much effort making this script Pythonic.